org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java

```diff
@@ -85,11 +88,10 @@ private void runAsJob(final String input) {
             @Override
             protected IStatus run(IProgressMonitor monitor) {
                 try {
-                    runAsUIJob(eval(input));
+                    IResult result = pool.submit(() -> eval(input)).get();
+                    runAsUIJob(result);
                     return Status.OK_STATUS;
-                } catch (CommandNotFoundException e) {
-                    StyledText message = new StyledText(Color.RED, e.getMessage());
-                    runAsUIJob((visitor) -> visitor.visitMessage(message));
+                } catch (InterruptedException | ExecutionException e) {
                     return Status.CANCEL_STATUS;
```

Comment:
Actually, the whole REPL should exit.

But first, please name a scenario in which this exception could happen.